### ISSUE TITLE
ROS discards credentials if they contain punctuation, username exists

### ISSUE THREAD

**Issue by icanzilb** [realm/realm-object-server#258]
## Goals

Create a user `eventblank_NSSpain@host` via the admin console and then login from an iOS app.

## Expected Results

Success.

## Actual Results

ROS rejects the credentials with the following message: `warn: auth.password: Handle request failed with: InvalidCredentials: The provided credentials are invalid or a user does not exist.: Given account: eventblank_NSSpain@host`

I tried many different variations and here's some outcome:

* `eventblank@host` can login just fine
* `eventblank_NSSpain@host` is rejected by ROS at auth time, the user exists and is created via the console
* `eventblank-NSSpain@host` is rejected by ROS at auth time, the user exists and is created via the console

I can't really say if the issue is on ROS side or iOS binding side but the end results is that the users can't use their emails as ROS usernames.

This issue _might_ be related to this (but also might not): https://github.com/realm/realm-mobile-platform/issues/159 - the only somewhat similar issue I found on this repo.

## Steps to Reproduce

Create `eventblank@host` and `eventblank_NSSpain@host` users and try to login with both from an iOS app. The latter will be rejected.

## Code Sample

n/a

## Version of Realm and Tooling

- Realm Object Server Version: 1.7.5 and 1.8.3
- Flavor:
  - [ ] Developer
  - [x] Professional
  - [ ] Enterprise
- Server OS & Version:  macOS 10.12.6 and Ubuntu 16.04.2 LTS
- Client SDK Version: 2.10
- Client OS & Version: macOS  10.12.6

## Logs

```
2017-08-25T06:39:38.525Z - info: sync: HTTP Connection[12]: Connection from 127.0.0.1:56207
2017-08-25T06:39:38.525Z - info: sync: HTTP Connection[12]: Received: Sync HTTP request(protocol_version=18)
2017-08-25T06:39:38.527Z - info: sync: Sync Connection[12]: Session[1]: Session initiated (session_ident=1).
2017-08-25T06:39:38.527Z - info: sync: Sync Connection[12]: Session[1]: Received: BIND(server_path=/f55a4e815f20f182896617e6b03870e1/eventblank, signed_user_token='...SGEk6DUbKD3ACrEBGfKdoxAQhISQ==', need_file_ident_pair=1)
2017-08-25T06:39:38.528Z - info: sync: Sync Connection[12]: Session[1]: Sending: ALLOC(server_file_ident=1, client_file_ident=3, client_file_ident_secret=488765936994759735)
2017-08-25T06:39:38.541Z - info: sync: Sync Connection[12]: Session[1]: Received: IDENT(server_file_ident=1, client_file_ident=3, client_file_ident_secret=488765936994759735, scan_server_version=0, scan_client_version=0, latest_server_version=0, latest_server_session_ident=0)
2017-08-25T06:39:38.541Z - info: sync: Sync Connection[12]: Session[1]: Received: MARK(request_ident=1)
2017-08-25T06:39:38.541Z - info: sync: Sync Connection[12]: Session[1]: Sending: MARK(request_ident=1)
2017-08-25T06:39:43.247Z - info: sync: Sync Connection[12]: Session[1]: Session terminated (session_ident=1).
2017-08-25T06:39:43.248Z - info: sync: Sync Connection[12]: Connection closed by client: End of input
2017-08-25T06:40:01.813Z - warn: auth.password: Handle request failed with: InvalidCredentials: The provided credentials are invalid or a user does not exist.: Given account: eventblank_NSSpain@host
```

**Comment by icanzilb** [comment]
Actually this is a duplicate of https://github.com/realm/realm-mobile-platform/issues/159 - the real problem seems to be the capital letters in the emails I've tried when looking how to repro

**Comment by DavidOliverDoswell** [comment]
I am having a similar issue but it has nothing to do with capitalized characters in my email addresses. I've created several accounts in the Realm Dashboard and they all give me the same error in Xcode's console when I log into my app : 

`Handle request failed with: InvalidCredentials: The provided credentials are invalid or a user does not exist.: Given account:` 

Here's my code:

` SyncUser.logIn(with: SyncCredentials.usernamePassword(username: username, password: password), server: URL(string: "http://************:9080")!) { (user, error) in
            
        guard let user = user else {
        fatalError(String(describing: error))
        }
            let permissions = SyncPermissionValue(realmPath: "*", userID: "*", accessLevel: .write)
            user.applyPermission(permissions) { error in
                if error != nil { return }
            }
            user.retrievePermissions { permissions, error in
                if error != nil { return }
            }
            
            DispatchQueue.main.async {
                let configuration = Realm.Configuration(
                    syncConfiguration: SyncConfiguration(user: user, realmURL: URL(string: "realm://************:9080/~/listRealm")!))
               
                let realm = try! Realm(configuration: configuration)
                self.items = realm.objects(Item.self).sorted(byKeyPath: "created", ascending: false)
                self.notificationToken = self.item.addNotificationBlock { [weak self] (changes: RealmCollectionChange) in
                    
                    guard (self?.collectionView) != nil else { return }
                    
                    switch changes {
                    case .initial:
                        self?.collectionView.reloadData()
                        break
                        
                    case .update(_, let deletions, let insertions, let modifications):
                        self?.collectionView.performBatchUpdates({
                            self?.collectionView.insertItems(at: insertions.map({ IndexPath(row: $0, section: 0)}))
                            self?.collectionView.deleteItems(at: deletions.map({ IndexPath(row: $0, section: 0)}))
                            self?.collectionView.reloadItems(at: modifications.map({ IndexPath(row: $0, section: 0)}))
                        }, completion: nil)
                        break
                        
                    case .error(let error):
                        print(error.localizedDescription)
                        break
                    }
                }
            }
        }`

I am able to `Connect to the Object Server` via the Mac app and can `Open Realm URL` with token and credentials. But I cannot log in from my iOS app. Any ideas?

**Comment by morten-krogh** [comment]
Thanks for reporting this issue. 
Indeed, there is a problem with certain characters in user names.

We will look into it.

**Comment by morten-krogh** [comment]
We will soon have a new release with this issue fixed.

**Comment by bigfish24** [comment]
This should be fixed with ROS 2.0
